src/field_map_change.c

```diff
@@ -610,41 +600,40 @@ static BOOL sub_02053718 (TaskManager * taskMan)
     return 0;
 }
 
-void sub_02053808 (FieldSystem * fieldSystem)
+void FieldSystem_StartLoadMapFromErrorTask (FieldSystem * fieldSystem)
 {
-    TaskManager * v0;
     UnkStruct_02053718 * v1;
 
-    if (MapHeader_IsUnionRoom(fieldSystem->unk_1C->unk_00)) {
-        (void)0;
-    } else if (sub_02053518(fieldSystem)) {
-        UnkStruct_020507E4 * v2 = SaveData_Events(fieldSystem->saveData);
+    if (MapHeader_IsUnionRoom(fieldSystem->location->mapId)) {
+        
+    } else if (FieldSystem_IsSaveInUnionRoom(fieldSystem)) {
+        VarsFlags * events = SaveData_GetVarsFlags(fieldSystem->saveData);
 
-        sub_02053540(fieldSystem);
-        sub_0206AD9C(v2);
+        FieldSystem_SetLocationToUnionRoomExit(fieldSystem);
+        sub_0206AD9C(events);
     } else {
-        sub_02053704(fieldSystem);
+        FieldSystem_SetLoadMapFromContinueGameTask(fieldSystem);
         return;
     }
```

Comment:
This is a case where I think I would rather we invert the sentinel (`MapHeader_IsUnionRoom`) and wrap the other blocks underneath it. Thoughts?